### PR TITLE
refactored main function

### PR DESCRIPTION
**Pull request by CodeConnoisseur74**
I refactored the main by creating a dictionary called 'actions. Each key corresponds to the user's choice in the menu and the value of the corresponding function. 

When the user makes a choice, 'actions.get(choice) is used to retrieve the function associated with that choice. 

I the user makes a valid choice, the function is called, otherwise,an error message is printed. 

Now each action is defined in it's own function (No more 'stairway of elifs').

I welcome your feedback :)

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        "4": delete_task_prompt,
+        "5": quit_program,
+    }
+
+    while True:
+        display_menu()
+        choice = input("Enter your choice: ")
+        action = actions.get(choice)
+
+        if action:
```
Comment (bbelderbos) [inline main.py:84]: I usually use `is` agains the None (singleton) if that is the return type (which is the case of dict.get here)

so you can do:

```
if action is not None:
    ...
```

You can also do an early return + continue here, so:

```
if action is None:
    print("...") # an altarnative is to raise an exception but let's not do that here
    continue

action()
```

File: main.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            action()
+        else:
+            print("Invalid choice. Please select a valid option.")
```
Comment (bbelderbos) [inline main.py:87]: maybe show the valid options in this message?

File: main.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+tasks = []
+
+
+# Main menu
```
Comment (bbelderbos) [inline main.py:6]: you can turn those comments into docstrings

File: main.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    print("2. View tasks")
+    print("3. Complete task")
+    print("4. Delete task")
+    print("5. Quit")
+
+
+# Add tasks to list
+def add_task():
+    task = input("Enter a task: ")
+    tasks.append({"task": task, "completed": False})
```
Comment (bbelderbos) [inline main.py:19]: there is still a bit of input + tasks manipulation mixed, but we can further break that out when we do the DB layer, I think `tasks.append`, `tasks[index]["completed"] = True` and `del tasks[index]` will then become functions on their own with the corresponding sqlmodel logic / code.

**Review by bbelderbos — APPROVED** [review]
Looks great, I like the dict dispatch pattern you've got going here :)
sqlmodel next!

File: main.py (hunk 1 above)
Comment (CodeConnoisseur74) [reply main.py:84]: Oh right! I never thought of that. I'll definitely try that! :)

File: main.py (hunk 2 above)
Comment (CodeConnoisseur74) [reply main.py:87]: Yes you're right! Always have to think of UX . So I could probably use: "Invalid choice. Please select ... then list the options."

File: main.py (hunk 3 above)
Comment (CodeConnoisseur74) [reply main.py:6]: Ok. Yes. :)

File: main.py (hunk 4 above)
Comment (CodeConnoisseur74) [reply main.py:19]: Oh Great:) Ok so I'll get the DB layer up and running on my own first. I'll prepare any questions I have for our coaching session. 

I'll send any errors I have here of course. 

Another thing I can do going forward: If I need help, I'll include the google search text I wrote, along with three resources. I really feel that my google searching could be a lot better as quite often, I end up down the rabbit hole. 

This way, you can give me tips on how to better look up resources, as well as the wording of my google search queries.

Comment (bbelderbos) [reply main.py:19]: Yes, providing the google search or ChatGPT thread will be useful to help you improve the asking questions part.

Yes please make a new branch with a Todo sqlmodel model and try to use this ORM to add the 4 required operations (add, show, update, delete). I think the thread I shared demos this but happy to provide more practical hints as we go ...

**Pull request merged**